internal/credentials/credentials.go

```diff
@@ -221,10 +231,40 @@ func (k *kubernetesDatabase) getCredentialsSecret(
 	return matchingSecret, nil
 }
 
-func secretToCreds(secret *corev1.Secret) Credentials {
+func (k *kubernetesDatabase) secretToCreds(
+	ctx context.Context, credType Type, secret *corev1.Secret,
+) (Credentials, error) {
+	if credType == TypeImage {
+		// If the cred type is image, we'll try to derive username and password
```

Comment:
Wonder if, at some point, it may be better to identify the type of credentials we are dealing with based on some user-defined value. Instead of the brute-force approach that's being taken now.